warning: ‘template<class> class std::auto_ptr’ is deprecated

compiling on Ubuntu 18.04 with gcc 7.3.0 gives:
```
/home/gijs/Work/casacore/ms/MSOper/test/tNewMSSimulator.cc:105:8: warning: ‘template<class> class std::auto_ptr’ is deprecated [-Wdeprecated-declarations]
   std::auto_ptr<casacore::NewMSSimulator> simulator_p;
        ^~~~~~~~
In file included from /usr/include/c++/7/memory:80:0,
                 from /home/gijs/Work/casacore/casacore/ms/MSOper/NewMSSimulator.h:33,
                 from /home/gijs/Work/casacore/ms/MSOper/test/tNewMSSimulator.cc:34:
/usr/include/c++/7/bits/unique_ptr.h:51:28: note: declared here
   template<typename> class auto_ptr;
                            ^~~~~~~~

```

--- a/casa/OS/Mutex.h
+++ b/casa/OS/Mutex.h
@@ -215,10 +215,22 @@ namespace casacore {
   {
   public:
     CallOnce0()
-    { }
+    {
+#ifndef USE_THREADS
+      itsFlag = True;
+#endif
+    }
 
     void operator()(void (*fn)()) {
+      //# call_once does not work properly without threads.
+#ifdef USE_THREADS
       std::call_once(itsFlag, fn);
+#else
+      if (itsFlag) {
+        fn();
+        itsFlag = False;
+      }
+#endif
     }
 
 private:
@@ -227,7 +239,11 @@ private:
     // Forbid assignment.
     CallOnce0& operator= (const CallOnce0&);
 
+#ifdef USE_THREADS
     std::once_flag itsFlag;
+#else
+    Bool itsFlag;
+#endif
   };
 
   // CallOnce: func has one arg.
@@ -236,11 +252,22 @@ private:
   {
   public:
     CallOnce()
-    { }
+    {
+#ifndef USE_THREADS
+      itsFlag = True;
+#endif
+    }
 
     template<typename T>
     void operator()(void (*fn)(T), T t) {
+#ifdef USE_THREADS
       std::call_once(itsFlag, fn, t);
+#else
+      if (itsFlag) {
+        fn(t);
+        itsFlag = False;
+      }
+#endif
     }
 
   private:
@@ -249,7 +276,11 @@ private:
     // Forbid assignment.
     CallOnce& operator= (const CallOnce&);
 
+#ifdef USE_THREADS
     std::once_flag itsFlag;
+#else
+    Bool itsFlag;
+#endif
   };
 
 } // namespace casacore

--- a/ms/MSOper/test/tNewMSSimulator.cc
+++ b/ms/MSOper/test/tNewMSSimulator.cc
@@ -102,7 +102,7 @@ public:
     nftw(msName_p.c_str(), removeFile, 64, FTW_DEPTH | FTW_PHYS);
   }
 
-  std::auto_ptr<casacore::NewMSSimulator> simulator_p;
+  std::unique_ptr<casacore::NewMSSimulator> simulator_p;
   std::string msName_p; 
 
 };

